### PR TITLE
[KYUUBI #4658] [Authz] [Bug] Fix InsertIntoHiveDirCommand classname so that we can extract the query in it when authorization.

### PR DESCRIPTION
**Pull request by Karsonnel**
### _Why are the changes needed?_

To fix https://github.com/apache/kyuubi/issues/4658.

### _How was this patch tested?_

Add ut that will run a InsertHiveDirCommand which query from a no permission table

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by Karsonnel** [comment]
@bowenliang123  can you help to review this pr? thank you very much.

**Review by yaooqinn — APPROVED** [review]

**Review by cxzl25 — APPROVED** [review]

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RangerSparkExtensionSuite.scala [hunk 1]
```diff
@@ -707,4 +707,21 @@ class HiveCatalogRangerSparkExtensionSuite extends RangerSparkExtensionSuite {
       sql(s"SHOW TABLES IN $db").queryExecution.optimizedPlan.stats
     }
   }
+
+  test("[KYUUBI #4658] INSERT OVERWRITE DIRECTORY did check query permission") {
+    val db1 = "default"
+    val table = "src"
+
+    withCleanTmpResources(Seq((s"$db1.$table", "table"))) {
+      doAs("bob", sql(s"CREATE TABLE IF NOT EXISTS $db1.$table (id int, name string)"))
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RangerSparkExtensionSuite.scala:716]: It's better to use `admin` for table preparation

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RangerSparkExtensionSuite.scala [hunk 2]
```diff
@@ -707,4 +707,21 @@ class HiveCatalogRangerSparkExtensionSuite extends RangerSparkExtensionSuite {
...
+      val e1 = intercept[AccessControlException](
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RangerSparkExtensionSuite.scala:717]: nit: e1 to e

**Comment by pan3793** [comment]
Thanks, merged to master/1.7

**Comment by yaooqinn** [comment]
@pan3793, it seems that the CI hasn't been finished yet, and some comments are still in progress?

File: extensions/spark/kyuubi-spark-authz/src/main/resources/table_command_spec.json [hunk 3]
```diff
@@ -1244,7 +1244,7 @@
     "fieldExtractor" : "LogicalPlanQueryExtractor"
   } ]
 }, {
-  "classname" : "org.apache.spark.sql.execution.datasources.InsertIntoHiveDirCommand",
```
Comment (bowenliang123) [inline extensions/spark/kyuubi-spark-authz/src/main/resources/table_command_spec.json:1247]: Can we make sure where or which spark version it comes from before removing support for the existed command removed from spec?

**Comment by pan3793** [comment]
@Karsonnel could you please open a follow-up PR to address reviewers' comments? thanks

**Comment by Karsonnel** [comment]
> @Karsonnel could you please open a follow-up PR to address reviewers' comments? thanks

sure, i'll do it.

File: extensions/spark/kyuubi-spark-authz/src/main/resources/table_command_spec.json (hunk 3 above)
Comment (Karsonnel) [reply extensions/spark/kyuubi-spark-authz/src/main/resources/table_command_spec.json:1247]: have checked spark version about 2.4, 3.1 and 3.2, i think it's a mistake of writing a wrong classname.

Comment (bowenliang123) [reply extensions/spark/kyuubi-spark-authz/src/main/resources/table_command_spec.json:1247]: OK, thanks for the check.

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RangerSparkExtensionSuite.scala (hunk 1 above)
Comment (bowenliang123) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RangerSparkExtensionSuite.scala:716]: Fix this minor problem as well ? @Karsonnel

**Pull request closed without merging**